### PR TITLE
Add confirmation dialog before importing contacts

### LINKED ISSUE #2788: Confirm before importing contacts
### Is your feature request related to a problem? Please describe.

When you click a `.vcf` file in the Files app, you are immediately transfered to the Contacts app which will start importing the VCF file. This can lead to undesired situations where you import a large amount of contacts into your contacts, either resulting in a mess or even duplicate contacts.

### Describe the solution you'd like

The app could show a confirmation window where you still need to click a button before it would start the import process. That also has the nice benefit that you could also select the destination contact list to import into.

### Describe alternatives you've considered

_No response_

### Additional context

Contacts 4.1.1, Nextcloud 24.0.1

### PR DESCRIPTION
**Pull request by GretaD**
fixes #2788

Screenshot from 2024-01-29 21-30-09
Screenshot from 2024-01-29 21-30-24

### CODE REVIEW

File: src/components/ConfirmationDialog.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+			<h2>{{ title }}</h2>
+			<slot />
+			<div class="confirm-modal__buttons">
+				<NcButton type="tertiary" :disabled="disabled" @click="cancel">
+					{{ t('contacts', 'Cancel') }}
+				</NcButton>
+				<NcButton v-if="resolve"
+					:href="confirmUrl"
+					:rel="confirmUrl ? 'noopener noreferrer' : false"
+					:target="confirmUrl ? '_blank' : false"
```
Comment (st3iny) [inline src/components/ConfirmationDialog.vue:13]:
```suggestion
```

File: src/components/ConfirmationDialog.vue [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			default: t('contacts', 'Confirm'),
+		},
+		confirmUrl: {
+			type: String,
+			default: undefined,
+		},
+		confirmAction: {
+			type: Function,
+			default: undefined,
+		},
```
Comment (st3iny) [inline src/components/ConfirmationDialog.vue:58]:
```suggestion
```

File: src/components/ConfirmationDialog.vue [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		disabled: {
+			type: Boolean,
+			default: undefined,
+		},
+	},
+	methods: {
+		confirm() {
+			this.resolve()
+			this.$emit('confirm')
```
Comment (st3iny) [inline src/components/ConfirmationDialog.vue:67]:
```suggestion
```

File: src/components/ConfirmationDialog.vue [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		},
+		cancel() {
+			this.reject()
+			this.$emit('cancel')
```
Comment (st3iny) [inline src/components/ConfirmationDialog.vue:71]:
```suggestion
```

**Review by st3iny — COMMENTED** [review]
Tested and works.

I left some annotations about code that is not required any more.

**Review by st3iny — APPROVED** [review]
Tested and works! Please squash.

File: src/files-action.js [hunk 5]
```diff
@@ -44,8 +49,33 @@ if (nextcloudVersionIsGreaterThanOr28) {
...
+							reject,
+						},
+					})
+					dialog.$mount(`#${containerId}`)
+				})
+
+				// Redirect to the import page if the user confirmed
+				window.location = generateUrl(`/apps/contacts/import?file=${file.path}`)
+			} catch (e) {
+				// Do nothing if the user cancels
```
Comment (ChristophWurst) [inline src/files-action.js:74]: logging would be advisable here if anything but rejection is thrown

**Pull request merged**